Dockerfile

```diff
@@ -1,13 +1,71 @@
-FROM node:12
+# ===============================================
+FROM helsinkitest/node:12-slim as appbase
+# ===============================================
+# Offical image has npm log verbosity as info. More info - https://github.com/nodejs/docker-node#verbosity
+ENV NPM_CONFIG_LOGLEVEL warn
 
-RUN mkdir -p /usr/src/app
+# set our node environment, either development or production
+# defaults to production, compose overrides this to development on build and run
+ARG NODE_ENV=production
+ENV NODE_ENV $NODE_ENV
 
-WORKDIR /usr/src/app
+# Global npm deps in a non-root user directory
+ENV NPM_CONFIG_PREFIX=/app/.npm-global
+ENV PATH=$PATH:/app/.npm-global/bin
 
-COPY package.json package.json
+# Yarn
+ENV YARN_VERSION 1.19.1
+RUN yarn policies set-version $YARN_VERSION
 
-COPY yarn.lock yarn.lock
+# Use non-root user
+USER appuser
 
-RUN yarn install
+# Copy package.json and package-lock.json/yarn.lock files
+COPY package*.json *yarn* ./
 
-CMD ["yarn", "start"]
\ No newline at end of file
+# Install npm depepndencies
+ENV PATH /app/node_modules/.bin:$PATH
+
+USER root
+
+RUN apt-install.sh build-essential
+
+USER appuser
+RUN yarn && yarn cache clean --force
+
+USER root
+RUN apt-cleanup.sh build-essential
+
+# =============================
+FROM appbase as development
+# =============================
+
+# Set NODE_ENV to development in the development container
+ARG NODE_ENV=development
+ENV NODE_ENV $NODE_ENV
+
+# copy in our source code last, as it changes the most
+COPY --chown=appuser:appuser . .
+
+# Bake package.json start command into the image
+CMD ["react-scripts", "start"]
+
+# ===================================
+FROM appbase as staticbuilder
+# ===================================
+
+ARG REACT_APP_API_URI
```

Comment:
All the given build arguments (`REACT_APP_*`) need to be declared here for them to be usable in `yarn build`.

Reply:
Like this?